### PR TITLE
CTCP-2447: Return Upscan-init details when attaching a large message to a departure

### PR DESCRIPTION
**Pull request by liandra-giacomine**

### CODE REVIEW

**Review by prashanthmamidi — APPROVED** [review]
resolve the conflicts

File: it/v2/connectors/PersistenceConnectorSpec.scala [hunk 1]
```diff
@@ -943,124 +943,208 @@ class PersistenceConnectorSpec
...
 
-        val future = persistenceConnector.postMessage(departureId, messageType, source).map(Right(_)).recover {
-          case NonFatal(e) => Left(e)
-        }
+          whenReady(future) {
+            result =>
+              result.left.toOption.get mustBe a[UpstreamErrorResponse]
+              val response = result.left.toOption.get.asInstanceOf[UpstreamErrorResponse]
+              response.statusCode mustBe INTERNAL_SERVER_ERROR
+              Json.parse(response.message).validate[StandardError] mustBe JsSuccess(StandardError("Internal server error", ErrorCode.InternalServerError))
```
Comment (drnaylor-oc) [inline it/v2/connectors/PersistenceConnectorSpec.scala:1115]: Not inherently wrong (though I think this throws warnings in Scala 2.13), but you may prefer something like:

```scala
whenReady(future) {
    case Left(UpstreamErrorResponse(`message`, INTERNAL_SERVER_ERROR, _,, _)) =>
              Json.parse(response.message).validate[StandardError] mustBe JsSuccess(StandardError("Internal server error", ErrorCode.InternalServerError))
    case other => fail(s"Expected internal server error, got $other")
}
```

I don't see this as a blocker though.

File: it/v2/connectors/PersistenceConnectorSpec.scala [hunk 2]
```diff
@@ -943,124 +943,208 @@ class PersistenceConnectorSpec
...
+            case NonFatal(e) => Left(e)
+          }
+
+          whenReady(future) {
+            result =>
+              result.left.toOption.get mustBe a[UpstreamErrorResponse]
+              val response = result.left.toOption.get.asInstanceOf[UpstreamErrorResponse]
+              response.statusCode mustBe BAD_REQUEST
+              Json.parse(response.message).validate[StandardError] mustBe JsSuccess(StandardError("Bad request", ErrorCode.BadRequest))
+          }
```
Comment (drnaylor-oc) [inline it/v2/connectors/PersistenceConnectorSpec.scala:1146]: Similar thing here.

File: test/v2/controllers/V2MovementsControllerSpec.scala [hunk 3]
```diff
@@ -4192,6 +4228,166 @@ class V2MovementsControllerSpec
...
+          arbitraryUpscanInitiateResponse.arbitrary
+        ) {
+          (movementId, messageId, upscanInitiateResponse) =>
+            beforeEach()
+
+            when(
+              mockPersistenceService
+                .addMessage(
+                  any[String].asInstanceOf[MovementId],
+                  any[MovementType],
```
Comment (drnaylor-oc) [inline test/v2/controllers/V2MovementsControllerSpec.scala:4251]: Do we know this? We should ensure that's what we're getting.

File: test/v2/controllers/V2MovementsControllerSpec.scala [hunk 4]
```diff
@@ -4192,6 +4228,166 @@ class V2MovementsControllerSpec
...
+                  any[Option[MessageType]],
```
Comment (drnaylor-oc) [inline test/v2/controllers/V2MovementsControllerSpec.scala:4252]: Same here -- is this something or nothing?

File: test/v2/controllers/V2MovementsControllerSpec.scala [hunk 5]
```diff
@@ -4192,6 +4228,166 @@ class V2MovementsControllerSpec
...
+            ).thenReturn(EitherT.rightT(UpdateMovementResponse(messageId)))
+
+            when(
+              mockUpscanService
+                .upscanInitiate(
+                  any[String].asInstanceOf[EORINumber],
+                  any[MovementType],
+                  any[String].asInstanceOf[MovementId],
+                  any[String].asInstanceOf[MessageId]
+                )(
```
Comment (drnaylor-oc) [inline test/v2/controllers/V2MovementsControllerSpec.scala:4267]: We probably know what all of these are meant to be?

File: app/v2/connectors/PersistenceConnector.scala [hunk 6]
```diff
@@ -212,6 +236,7 @@ class PersistenceConnectorImpl @Inject() (httpClientV2: HttpClientV2, appConfig:
     httpClientV2
       .patch(url"$url")
       .withBody(Json.toJson(body))
+      .setHeader(Constants.XMessageTypeHeader -> messageType.code)
```
Comment (drnaylor-oc) [inline app/v2/connectors/PersistenceConnector.scala:239]: The patch call doesn't need this as the message type will be in the body (but it doesn't harm keeping it).

**Review by drnaylor-oc — APPROVED** [review]
Minor comment that you don't need to do anything about.

**Pull request merged**